lib/internal/loader/ModuleRequest.js

```diff
@@ -36,24 +38,39 @@ loaders.set('esm', async (url) => {
 
 // Strategy for loading a node-style CommonJS module
 loaders.set('cjs', async (url) => {
-  return createDynamicModule(['default'], url, (reflect) => {
-    debug(`Loading CJSModule ${url}`);
-    const CJSModule = require('module');
-    const pathname = internalURLModule.getPathFromURL(new URL(url));
-    CJSModule._load(pathname);
+  debug(`Loading CJSModule ${url}`);
+  const CJSModule = require('module');
+  const pathname = internalURLModule.getPathFromURL(new URL(url));
+  const exports = CJSModule._load(pathname);
+  const es = exports[esModuleInterop] !== undefined ?
+    exports[esModuleInterop] : exports.__esModule;
+  const keys = es ? Object.keys(exports) : ['default'];
+  return createDynamicModule(keys, url, (reflect) => {
+    if (es) {
```

Comment:
Is branching like this faster? Otherwise seems like the else branch does exactly what the `if` branch does, just more generally.

Reply:
i don't understand what you mean

Reply:
Sign, nevermind, just another case of misreading.